Library/brew.rb

```diff
@@ -98,7 +98,7 @@ def require?(path)
   cmd = HOMEBREW_INTERNAL_COMMAND_ALIASES.fetch(cmd, cmd)
 
   sudo_check = %w[ install reinstall postinstall link pin unpin
-                   update upgrade create migrate tap switch ]
+                   update upgrade create migrate tap switch fix-permissions ]
```

Comment:
Did some local tests. I think current `sudo_check` behavior is exactly what we want.
```
$ ruby -e 'puts File.owned?("/usr/local/bin/brew")'
true

$ sudo ruby -e 'puts File.owned?("/usr/local/bin/brew")'
false

$ ruby -e 'puts ENV["USER"]'
xucheng

$ sudo ruby -e 'puts ENV["USER"]'
root
```